### ISSUE TITLE
The default setting of use_gt_layout is None in train_vqa_rl_gt_layout.py

### ISSUE THREAD

**Issue by happylicio** [ronghanghu/n2nmn#5]
https://github.com/ronghanghu/n2nmn/blob/f7a6e6913f1830d70bbcc9df1b9288809fc24321/exp_vqa/train_vqa_rl_gt_layout.py#L90

Hi:

I find that your setting of use_gt_layout is None while defining an instance of  NMN2Model in train_vqa_rl_gt_layout.py, which leads to training a layout policy without groud-truth layout. Although your default pretrained model from train_vqa_rl_gt_layout uses groud-truth layout, why isn't this training the same? 

What I understand so far is that we set use_gt_layout to None when we need to evaluate our model on the test dataset. Can you explain your considerations for the use_gt_layout setting? 

Best
Licio

**Comment by ronghanghu** [comment]
Hi,

At test time `use_gt_layout` is set to none as ground-truth layouts are not available (and should not be used) in the test set.

During training, for train_vqa_rl_gt_layout.py, setting `use_gt_layout` to None allows the model to optimize the layout policy by performing policy search over the space, without constraining it to be the same as the ground-truth layout. In this scenario, the layout policy is optimized via policy gradient (REINFORCE).

**Comment by happylicio** [comment]
Hi, 

Thanks for your timely reply. After re-reading your paper, the ground-truth layout only used as a good initialization in training the layout policy.  

So I'm curious about the VQA test-dev accuracy of **policy search without cloning**(i.e.  we only define the layout search space and then train from scratch). Have you already done this experiment?

![image](https://user-images.githubusercontent.com/18717873/32988066-a9e9d342-cd36-11e7-96f5-f0f86e78ead7.png)

**Comment by ronghanghu** [comment]
Hi, I haven't tried training on the VQA dataset entirely from scratch, and don't have results for that.
